### PR TITLE
Feature/fga/room list filter iteration

### PR DESCRIPTION
**Pull request by ganfra**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [X] Other :

## Content
Iterate on the room list filters and refactor room list content state management.

List of changes : 

- Filter orderings (Unread, People, Room, Favourite)
- Filter animation 
- Filter integration with TopBar and Bloom
- Filter empty states
- RoomList content management (Skeleton, Empty, Migration, Rooms). 

## Motivation and context
Closes https://github.com/element-hq/element-x-android/issues/2486
Closes https://github.com/element-hq/element-x-android/issues/2463
Closes https://github.com/element-hq/element-x-android/issues/2462

## Screenshots / GIFs
In PR.

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/RoomListContentStateProvider.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import io.element.android.features.roomlist.impl.model.RoomListRoomSummary
+import kotlinx.collections.immutable.ImmutableList
+
+open class RoomListContentStateProvider : PreviewParameterProvider<RoomListContentState> {
+    override val values: Sequence<RoomListContentState>
+        get() = sequenceOf(
+            aRoomsContentState(),
+            aSkeletonContentState(),
+            anEmptyContentState(),
+            aMigrationContentState(),
```
Comment (jmartinesp) [inline features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/RoomListContentStateProvider.kt:29]: 👌

File: tests/testutils/src/main/kotlin/io/element/android/tests/testutils/MutablePresenter.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+package io.element.android.tests.testutils
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import io.element.android.libraries.architecture.Presenter
+import kotlinx.coroutines.flow.MutableStateFlow
+
+class MutablePresenter<State>(initialState: State) : Presenter<State> {
```
Comment (jmartinesp) [inline tests/testutils/src/main/kotlin/io/element/android/tests/testutils/MutablePresenter.kt:24]: Oh, nice way to simplify tests with composite presenters.

**Review by jmartinesp — APPROVED** [review]
Thanks, LGTM! The filters seem so much better now.

The only thing I found odd is the localazy string updates. Shouldn't these be already part of `develop`? I remember changing those myself for a previous PR.

**Comment by ganfra** [comment]
> Thanks, LGTM! The filters seem so much better now.
> 
> The only thing I found odd is the localazy string updates. Shouldn't these be already part of `develop`? I remember changing those myself for a previous PR.

I had started my branch before it landed on develop, and then got some conflict with room moderation strings. Benoit made some changes again, so I'll have fix it!

**Comment by bmarty** [comment]
image

:(

**Comment by ganfra** [comment]
> image
> :(
> 
> Can you check how which tests are missing?

I've added a bit more tests, but there is still a small drop. Can I handle that in another PR? This one is a bit long now, and I don't want to run again the CI for 3 hours

**Pull request merged**